app/services/creator_analytics/churn.rb

```diff
@@ -0,0 +1,263 @@
+# frozen_string_literal: true
+
+# Calculates customer churn rate metrics for subscription products.
+#
+# Churn rate formula (per Stripe):
+#   (Churned Customers / Total Base Customers) × 100
+#
+# Where Total Base = Active at Start + New During Period
+#
+# @see https://stripe.com/resources/more/monthly-churn-101
+class CreatorAnalytics::Churn
+  include ActiveModel::Validations
+
+  DEFAULT_START_DATE_OFFSET = 1.month
+  CACHE_EXPIRY = 24.hours
+
+  validates :end_date, comparison: { greater_than_or_equal_to: :start_date }
+
+  attr_reader :start_date, :end_date, :user, :products
+
+  def initialize(user:, start_date: nil, end_date: nil, params: {}, products: nil)
+    @user = user
+    @params = params
+    @start_date = parse_date(start_date, :start)
+    @end_date = parse_date(end_date, :end)
+    @products = products || parse_products
+  end
+
+  def fetch_churn_data
+    return nil unless has_subscription_products?
+
+    if should_use_cache?
+      fetch_cached_data
+    else
+      fetch_realtime_data
+    end
+  end
+
+  def has_subscription_products?
+    user.products.alive.is_recurring_billing.exists?
+  end
+
+  def available_products
+    user.products_for_creator_analytics.select(&:is_recurring_billing?)
+  end
+
+  def time_window
+    (end_date - start_date).to_i + 1
+  end
+
+  def self.customer_churn_rate(user:, start_date:, end_date:, products: nil)
+    new(
+      user: user,
+      start_date: start_date,
+      end_date: end_date,
+      products: products
+    ).send(:calculate_churn_rate)
+  end
+
+  private
+    def fetch_realtime_data
+      raise ArgumentError, "Invalid date range: #{errors.full_messages.join(', ')}" unless valid?
+
+      subs = subscriptions_for_period
+      period_metrics = calculate_period_metrics(start_date, end_date, subs)
+      daily_results = build_daily_results(subs)
+      last_period_rate = calculate_last_period_churn_rate
+
+      {
+        start_date: start_date.to_s,
+        end_date: end_date.to_s,
+        metrics: build_metrics(period_metrics, last_period_rate),
+        daily_data: format_daily_data(daily_results)
+      }
+    end
+
+    def fetch_cached_data
+      Rails.cache.fetch(cache_key, expires_in: CACHE_EXPIRY) do
+        fetch_realtime_data
+      end
+    end
+
+    def should_use_cache?
+      LargeSeller.where(user: user).exists?
+    end
+
+    def cache_key
+      @cache_key ||= begin
+        product_ids = if @products.respond_to?(:map)
+          @products.map(&:id).sort.join(",")
+        else
+          "all"
+        end
+        "seller_daily_churn_metrics:#{user.id}:#{start_date}:#{end_date}:#{product_ids}"
+      end
+    end
+
+    def subscriptions_for_period
+      @subscriptions_for_period ||= fetch_subscriptions(from: start_date, to: end_date).load
+    end
+
+    def build_daily_results(subscriptions)
+      (start_date..end_date).map do |date|
+        metrics = calculate_period_metrics(date, date, subscriptions)
```

Comment:
Use the same method for per-day calculations as well - it calculates for the period of one day only.